bin/build-vendor

```diff
@@ -1,10 +0,0 @@
-#!/usr/bin/env bash
-
-set -e
-
-cd "${BASH_SOURCE%/*}/.."
-
-rm -rf build-vendor
-
-COMPOSER_VENDOR_DIR=build-vendor composer update \
-  --prefer-stable --no-dev --no-progress --classmap-authoritative --no-interaction --verbose
```

Comment:
This does a few things that aren't captured in the current makefile:

1. uses a separate composer root, to ensure that the vendor directory is clean (and to build without screwing with the user's current vendor directory)
2. prefers stable, doesn't install dev dependencies
3. uses an optimized autoloader and classmap-authoritative to put more of the burden of autoloading on build time rather than runtime

Reply:
I'm not sure to follow 1, but 2 & 3 are done by Box

Reply:
Updated the makefile for it to never change whatever you currently have in `composer.json`, `composer.lock` or `vendor`